data-raw/healthy-lives/sexual-health.R

```diff
@@ -0,0 +1,139 @@
+# ---- Load packages ----
+library(tidyverse)
+library(readxl)
+library(httr)
+library(geographr)
+library(demographr)
+
+
+# ---- Get and clean data ----
+# Wales LTLA and HB Codes
+wales_hb_ltla <- lookup_ltla21_lhb22
+
+# Wales Latest LA Population Data
+wales_population <- population20_ltla21 |>
+  filter(str_starts(ltla21_code, "W")) |>
+  select(
+    ltla21_code,
+    ltla21_name,
+    total_population
+  )
+
+# Sexual Health data
+# Source: https://phw.nhs.wales/publications/publications1/sexual-health-trends-in-wales-annual-report-2024/
+sexual_health_raw <- read_excel("data-raw/healthy-lives/raw-data/sexual_health.xlsx")
+
+
+# SEXUAL HEALTH TESTING RATE
+testing_number <- sexual_health_raw |>
+  slice(12:19) |>
+  mutate(`...2` = case_when(
+    `...2` == "ABUHB" ~ "Aneurin Bevan University Health Board",
+    `...2` == "BCUHB" ~ "Betsi Cadwaladr University Health Board",
+    `...2` == "CTMUHB" ~ "Cwm Taf Morgannwg University Health Board",
+    `...2` == "CVUHB" ~ "Cardiff and Vale University Health Board",
+    `...2` == "HDUHB" ~ "Hywel Dda University Health Board",
+    `...2` == "PTB" ~ "Powys Teaching Health Board",
+    `...2` == "SBUHB" ~ "Swansea Bay University Health Board",
+    TRUE ~ `...2`
+  )) |>
+  filter(`...2` != "Unknown")
+
+# Calculate rate per 100,000
+testing_rate <- testing_number |>
+  left_join(wales_hb_ltla, by = c("...2" = "lhb22_name")) |>
+  left_join(wales_population) |>
+  group_by(ltla21_code) |>
+  mutate(testing_rate_per_100k = (`2023` / `total_population`) * 100000) |>
+  ungroup() |>
+  select(
+    ltla24_code = ltla21_code,
+    testing_rate_per_100k
+  )
```

Comment:
I'm not sure this method is valid (but I'm struggling to articulate to myself why!). Can you explain what your thinking was around this approach?

Think of this silly scenario:
- We have a healthboard called "x" with a testing number of 2
- We have a LTLA called "y" with a population of 2
- We have a LTLA caled "z" with a population of 4

Using the logic presented here:
- "y" would have a testing rate of 2/2 = 100 %
- "z" would have a testing rate of 2/4 = 50 %
- This means we would expect 4 people to have been detected in the testing number (2 from "y", i.e., the whole population, and 2 from "z", half the population), yet we know only 2 people were tested. Note we are ignorinig the scaling factor of 100,000 (as it simplifis the example).

Reply:
I think without knowing which LTLA's people came from, we have to assume an equal rate among all of them.

Reply:
@matthewgthomas not sure if you also have any reflections on the above?

Reply:
Is the issue that number of tests is for Health Boards but we need to spread that across LAs? Would it make sense to just spread it equally across LAs within a Health Board, if so?

Reply:
I don't think so, because surely larger local authorities will have a higher **_number_** of positive tests. But what we can't say is that they will have a higher _**rate**_ of positive tests.

Reply:
Going back to our problem, let's say we have our health board called x, and local authorities called y and z, which are covered by the health board, I _think_ we can define the problem more generally as:

$$ \frac{\text{y count}}{\text{y population}} + \frac{\text{z count}}{\text{z population}} = \text{x count} $$

This will always leave us with an unsolvable linear equation where we can't compute the counts for the populations if there is more than one local authority (in this case "y count" and "z count").